voting_schemes/electionguard/python-wrapper/src/bulletin_board/electionguard/bulletin_board.py

```diff
@@ -1,46 +1,274 @@
-from collections import defaultdict
-from typing import Dict, NoReturn, Optional, Set, Literal, Union, Tuple, List
+from typing import Any, Dict, List, Literal, NoReturn, Optional, Set, Tuple, Union
+
 from electionguard.ballot import (
+    BallotBoxState,
     CiphertextBallot,
+    SubmittedBallot,
     from_ciphertext_ballot,
-    BallotBoxState,
 )
 from electionguard.ballot_validator import ballot_is_valid_for_election
-from electionguard.decrypt_with_shares import decrypt_selection_with_decryption_shares
-from electionguard.election import ElectionConstants
-from electionguard.key_ceremony import combine_election_public_keys, ElectionPublicKey
-from electionguard.tally import (
-    CiphertextTally,
-    PlaintextTallySelection,
-    PlaintextTallyContest
+from electionguard.data_store import DataStore
+from electionguard.decrypt_with_shares import decrypt_tally
+from electionguard.decryption import (
+    AVAILABLE_GUARDIAN_ID,
+    MISSING_GUARDIAN_ID,
+    reconstruct_decryption_share,
 )
+from electionguard.decryption_share import CompensatedDecryptionShare, DecryptionShare
+from electionguard.election import ElectionConstants
+from electionguard.group import ElementModQ
+from electionguard.key_ceremony import ElectionPublicKey, combine_election_public_keys
+from electionguard.tally import CiphertextTally, PlaintextTally, PlaintextTallyContest
 from electionguard.types import CONTEST_ID, GUARDIAN_ID, SELECTION_ID
 from electionguard.utils import get_optional
+
 from .common import Content, Context, ElectionStep, Wrapper
 from .messages import (
+    Compensation,
+    KeyCeremonyResults,
     TrusteeElectionKey,
     TrusteePartialKeys,
     TrusteeVerification,
-    TrusteeShare,
-    KeyCeremonyResults
-)
-from .utils import (
-    InvalidBallot,
-    serialize,
-    deserialize,
 )
-from .dummy_scheduler import DummyScheduler
+from .utils import InvalidBallot, deserialize, serialize
+
+TallyResults = Dict[CONTEST_ID, Dict[SELECTION_ID, int]]
+
+
+class TallyDecryptor:
```

Comment:
💬 This class is pretty big, maybe we could move it to a separate file?

Reply:
Extracted~